### PR TITLE
refactor(samples): Refactor the delete digital twin sample snippet

### PR DESCRIPTION
**Pull request by abhipsaMisra**

### CODE REVIEW

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs [hunk 1]
```diff
@@ -217,12 +219,18 @@ await foreach (IncomingRelationship incomingRelationship in incomingRelationship
...
                     #region Snippet:DigitalTwinsSampleDeleteTwin
 
-                    Response deleteDigitalTwinResponse = await client.DeleteDigitalTwinAsync(digitalTwinId);
-                    Console.WriteLine($"Deleted digital twin with Id {digitalTwinId}. Response Status: {deleteDigitalTwinResponse.Status}");
+                    try
+                    {
+                        Response deleteDigitalTwinResponse = await client.DeleteModelAsync(digitalTwinId);
+                        Console.WriteLine($"Deleted digital twin with Id {digitalTwinId}. Response Status: {deleteDigitalTwinResponse.Status}");
+                    }
+                    catch (Exception ex)
```
Comment (azabbasi) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs:230]: Shouldn't we just catch RequestFailedException ?

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs [hunk 2]
```diff
@@ -217,12 +219,18 @@ await foreach (IncomingRelationship incomingRelationship in incomingRelationship
                     }
 
                     // Now the digital twin should be safe to delete
-                    string digitalTwinId = twin.Key;
 
                     #region Snippet:DigitalTwinsSampleDeleteTwin
 
-                    Response deleteDigitalTwinResponse = await client.DeleteDigitalTwinAsync(digitalTwinId);
-                    Console.WriteLine($"Deleted digital twin with Id {digitalTwinId}. Response Status: {deleteDigitalTwinResponse.Status}");
+                    try
```
Comment (azabbasi) [inline sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs:225]: I think the reason behind the big try catch block was to remove the try catch from the snippet region, let's stick to that if possible.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs (hunk 1 above)
Comment (abhipsaMisra) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs:230]: yes, good point. I'll update it.

File: sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs (hunk 2 above)
Comment (abhipsaMisra) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs:225]: sure

Comment (azabbasi) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs:225]: correct me if I am wrong here @drwill-ms

Comment (abhipsaMisra) [reply sdk/digitaltwins/Azure.DigitalTwins.Core/samples/DigitalTwinsClientSample/DigitalTwinsLifecycleSamples.cs]: I agree, doesn't make sense to enclose a try-catch within another try-catch block, for our scenario.

**Review by barustum — APPROVED** [review]

**Pull request merged**